dillo/web/users/routes.py

```diff
@@ -33,18 +34,26 @@ def users_view(username):
     if user is None:
         return abort(404)
     api = system_util.pillar_api()
-    posts = Node.all({
-        'where': {'user': str(user['_id']), 'node_type': 'dillo_post', 'properties.status': 'published'},
-        'sort': '-_created',
-    }, api=api)
+    nodes_coll = current_app.db('nodes')
 
-    for post in posts['_items']:
-        if post.picture:
-            post.picture = get_file(post.picture, api=api)
+    pipeline = [
+        {'$match': {'user': ObjectId(user['_id']), 'node_type': 'dillo_post', 'properties.status': 'published'}},
+        {'$lookup': {
+            'from': 'projects',
+            'localField': 'project',
+            'foreignField': '_id',
+            'as': 'project'}},
```

Comment:
Add a projection:

```
        {'$project': { 
            'name': 1, 
            'properties': 1, 
            'user': 1, 
            'picture': 1, 
            '_created': 1, 
            'project': {'$arrayElemAt': ['$project', 0]} 
        }}, 
```

This will force 1 project per project field (and not an array).